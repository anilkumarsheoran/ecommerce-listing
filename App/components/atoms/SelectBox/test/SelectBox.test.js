import { shallow } from "enzyme";
import { SelectBox } from "../SelectBox";
const { configure } = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

configure({ adapter: new Adapter() });

const props = {
  options: [
    {
      label: "Filter By Size",
      value: ""
    },
    {
      label: "XS",
      value: "XS"
    }
  ],
  selected: "S"
};

describe("SelectBox Component", () => {
  test("should render correctly", () => {
    const selectbox = shallow(<SelectBox {...props} />);
    expect(selectbox).toMatchSnapshot();
  });

  test("should call onChange", () => {
    const selectbox = shallow(<SelectBox {...props} />);
    const event = {
      target: { value: "X" }
    };
    selectbox.find("select").simulate("change", event);
    expect(selectbox.state().value).toBe("X");
  });
});
