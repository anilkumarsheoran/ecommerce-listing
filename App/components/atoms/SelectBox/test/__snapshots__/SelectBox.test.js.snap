// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectBox
    className=""
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "Filter By Size",
          "value": "",
        },
        Object {
          "label": "XS",
          "value": "XS",
        },
      ]
    }
    selected="S"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="filter-label"
          htmlFor="select-box"
        >
          Select
        </label>,
        <select
          className="form-control"
          id="select-box"
          onChange={[Function]}
          value="S"
        >
          <option
            value=""
          >
            Filter By Size
          </option>
          <option
            value="XS"
          >
            XS
          </option>
        </select>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select",
          "className": "filter-label",
          "htmlFor": "select-box",
        },
        "ref": null,
        "rendered": "Select",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value=""
            >
              Filter By Size
            </option>,
            <option
              value="XS"
            >
              XS
            </option>,
          ],
          "className": "form-control",
          "id": "select-box",
          "onChange": [Function],
          "value": "S",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "",
            "nodeType": "host",
            "props": Object {
              "children": "Filter By Size",
              "value": "",
            },
            "ref": null,
            "rendered": "Filter By Size",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "XS",
            "nodeType": "host",
            "props": Object {
              "children": "XS",
              "value": "XS",
            },
            "ref": null,
            "rendered": "XS",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="filter-label"
            htmlFor="select-box"
          >
            Select
          </label>,
          <select
            className="form-control"
            id="select-box"
            onChange={[Function]}
            value="S"
          >
            <option
              value=""
            >
              Filter By Size
            </option>
            <option
              value="XS"
            >
              XS
            </option>
          </select>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select",
            "className": "filter-label",
            "htmlFor": "select-box",
          },
          "ref": null,
          "rendered": "Select",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value=""
              >
                Filter By Size
              </option>,
              <option
                value="XS"
              >
                XS
              </option>,
            ],
            "className": "form-control",
            "id": "select-box",
            "onChange": [Function],
            "value": "S",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "",
              "nodeType": "host",
              "props": Object {
                "children": "Filter By Size",
                "value": "",
              },
              "ref": null,
              "rendered": "Filter By Size",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "XS",
              "nodeType": "host",
              "props": Object {
                "children": "XS",
                "value": "XS",
              },
              "ref": null,
              "rendered": "XS",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
